File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Acquire Sync Sample/AcquireSyncSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.AddIn
Imports System.AddIn.Pipeline
Imports PrincetonInstruments.LightField.AddIns

''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'''''''''' This sample acquires a single frame of data synchronously.                               ''''''''''
'''''''''' When the data is returned, pop it into a window (experiment view 1) for user to look at. ''''''''''
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''' 
Public Class AcquireSyncSample
    Inherits AddInBase

    Private control_ As SyncAcquireSample

    Public ReadOnly Property UISupport As UISupport
        Get
            Return UISupport.ExperimentSetting
        End Get
    End Property

    Public Sub Activate(ByVal app As ILightFieldApplication)
        'Capture interface
        LightFieldApplication = app
        'Build your controls
        control_ = New SyncAcquireSample(app)
        ExperimentSettingElement = control_
    End Sub

    Public Sub Deactivate()
    End Sub

    Public Shadows ReadOnly Property UIExperimentSetting As Windows.FrameworkElement
        Get
            If ExperimentSettingElement Is Nothing Then
                Throw New NotImplementedException("UIExperiment has no implementation!")
            Else
                Return ExperimentSettingElement
            End If
        End Get
    End Property

    Public Overrides ReadOnly Property UIExperimentSettingTitle As String
        Get
            Return "Acquire Synchronous Sample"
        End Get
    End Property

End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Acquire Sync Sample/AcquireSyncSample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.AddIn;
using System.AddIn.Pipeline;

using PrincetonInstruments.LightField.AddIns;

namespace LightFieldAddInSamples
{
    ///////////////////////////////////////////////////////////////////////////
    // This sample acquires a single frame of data synchronously and when the 
    // data is returned it pops it into a window (experiment view 1) for the 
    // user to look at.
    ///////////////////////////////////////////////////////////////////////////
    [AddIn("Acquire Synchronous Sample (C#)", 
    Version = "1.0.0",
    Publisher = "Teledyne Princeton Instruments",
    Description = "Shows an example of synchronous acquisition")]
    [QualificationData("IsSample", "True")]
    public class AddinAcquireSyncSample : AddInBase, ILightFieldAddIn
    {
        private Acquire_Sync_Sample.AcquireSyncSample control_;
        ///////////////////////////////////////////////////////////////////////
        public UISupport UISupport { get { return UISupport.ExperimentSetting; } }
        ///////////////////////////////////////////////////////////////////////
        public void Activate(ILightFieldApplication app)
        {
            // Capture Interface
            LightFieldApplication = app;

            // Build your controls
            control_ = new Acquire_Sync_Sample.AcquireSyncSample(app);            
            ExperimentSettingElement = control_;
        }
        ///////////////////////////////////////////////////////////////////////
        public void Deactivate() { }
        ///////////////////////////////////////////////////////////////////////        
        public override string UIExperimentSettingTitle { get { return "Acquire Synchronous Sample"; } }
    }
}
